File: backend/api/schemas/session.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CreateSessionRequest(BaseModel):
    user_id: str
    max_rounds: int = 5
    metadata: dict = {}


class UpdateSessionRequest(BaseModel):
    status: Optional[str] = None
    max_rounds: Optional[int] = None


class MedicalRecordSummary(BaseModel):
    session_id: Optional[str] = None
    version: int = 0
    completion_level: str = "partial"
    chief_complaint: Optional[str] = None
    duration: Optional[str] = None
    location: Optional[str] = None
    severity: Optional[int] = None
    accompanying_symptoms: list[str] = []
    collected_fields: list[str] = []
    missing_core_fields: list[str] = []


class SessionResponse(BaseModel):
    id: str
    user_id: str
    status: str
    intent: Optional[str] = None
    current_stage: Optional[str] = None
    red_flag_raised: bool = False
    round_count: int = 0
    max_rounds: int = 5
    close_reason: Optional[str] = None
    closed_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    medical_record_summary: Optional[MedicalRecordSummary] = None
    message_count: int = 0

File: frontend/src/types/session.ts
/** 会话状态 */
export type SessionStatus = 'active' | 'paused' | 'completed' | 'emergency_terminated' | 'closed_timeout'

/** 工作流阶段 */
export type WorkflowStage = 'init' | 'collect' | 'diagnose' | 'done' | 'emergency'

/** 会话意图 */
export type SessionIntent = 'diagnosis' | 'question' | 'emergency' | 'greeting' | 'status' | 'follow_up'

/** 会话基本信息（列表项） */
export interface Session {
  id: string
  session_id: string
  user_id?: string
  title?: string
  status: SessionStatus | WorkflowStage
  current_stage: WorkflowStage
  intent?: SessionIntent
  red_flag_raised: boolean
  round_count: number
  max_rounds: number
  close_reason?: string | null
  closed_at?: string | null
  created_at?: string
  updated_at?: string
}

/** 会话详情 */
export interface SessionDetail extends Session {
  medical_record_summary?: MedicalRecordSummary
  message_count?: number
}

/** 问诊记录摘要 */
export interface MedicalRecordSummary {
  session_id: string
  version: number
  completion_level: 'partial' | 'core_complete' | 'full'
  chief_complaint?: string
  duration?: string
  location?: string
  severity?: number
  accompanying_symptoms?: string[]
  collected_fields?: string[]
  missing_core_fields?: string[]
}
